endpoints/openrtb2/video_auction.go

```diff
@@ -279,6 +279,21 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 		bidResp.Ext = response.Ext
 	}
 
+	if len(bidResp.AdPods) == 0 && debugLog.Enabled {
+		err := debugLog.PutDebugLogError(deps.cache, vo.Errors)
+		if err != nil {
+			vo.Errors = append(vo.Errors, err)
+		} else {
+			bidResp.AdPods = append(bidResp.AdPods, &openrtb_ext.AdPod{
```

Comment:
Question: Would we want to return this `AdPod` with debug key only when we have 0 `AdPods` in the `bidResp`? What about the case when we get an error during any of the previous steps like while holding the auction or while building the response? In those cases we don't want to send the `AdPod` with debug key back?

Also, how will the user be able to differentiate between a normal bid response with an actual ad pod vs a bid response with a debug ad pod?

Reply:
For the first question, the error cases should be critical errors, which only return a message rather than a response with pods. So I kept the same error behavior for those failures, but the cache key is appended to the error message for log checking.

For the second question, the differentiator for the debug log `AdPod` is that it will not contain an `hb_pb` and `hb_pb_cat_dur`, and the `podid` will be 0.

Reply:
The error message usage is meant to cover error cases like a missing field in the request or unmarshalling errors, where we can only return a critical error message. What was missed initially was the case where there weren't any critical errors, but there weren't any bids either. The creating of the debug pod is meant to cover this case, while the error message usage will cover when the request fails for some reason and we need to see the cache ID.

The cache key is appended to the error message in `video_auction.go::handleError()`.

Reply:
The error message usage is meant to cover error cases like a missing field in the request or unmarshalling errors, where we can only return a critical error message. What was missed initially was the case where there weren't any critical errors, but there weren't any bids either. The creating of the debug pod is meant to cover this case, while the error message usage will cover when the request fails for some reason and we need to see the cache ID.

The cache key is appended to the error message in video_auction.go::handleError().